content/docs/tutorials/handling-errors.mdx

```diff
@@ -0,0 +1,243 @@
+
+  - **Adjusting the request** to resolve structural errors with queries or transaction submissions is the first line of defense: if you've included a bad parameter, malformed your XDR, or otherwise didn't follow the endpoint's specification, the error can be resolved by referencing the details or result codes of the error response.
+  - **Retrying until success** is the recommended way to work around latency or congestion issues encountered anywhere along the pipeline between your machine and the Stellar network. This ephemeral scenario is [unavoidable](https://en.wikipedia.org/wiki/High_availability#%22Nines%22) due to the very nature of a distributed system.
+  - **Adjusting the transaction** can also resolve issues but it should only be done with *extreme* care: if one of the above scenarios is in effect, it's possible to trigger destructive duplicate actions (like sending a payment twice).
+
+Let's dive into these strategies in detail. The main scenario we'll focus on is transaction submission, since it's an action with meaningful side-effects rather than a read-only request.
+
+
+### Request Adjustments
+
+> We cannot direct the wind, but we can adjust the sails.  
+
+Some errors cannot be overcome without changes to the request itself. 
+
+#### Queries
+Many of the `GET` requests have specific parameter requirements, and while the [SDKs](../software-and-sdks/) can help enforce them, you can still pass invalid arguments (e.g. an asset string that isn't [SEP-11 conformant](https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0011.md#asset)) that error out every time. In this scenario, there's nothing you can do aside from following the [API specification](../../api) precisely. The `extras` field of the error response will often clue you in on where to look and what to look for:
+
+<CodeExample>
+
+```bash
+curl -s https://horizon-testnet.stellar.org/claimable_balances/0000 | jq '.extras'
+{
+  "invalid_field": "id",
+  "reason": "Invalid claimable balance ID"
+}
+```
+
+</CodeExample>
+
+Note that the SDKs make it a point to distinguish an **invalid request** (as above) vs. a **missing resource** (a `404 Not Found`) (e.g. the generic `NetworkError` vs. a `NotFoundError` in the JavaScript SDK), where the latter might not be considered an error depending on your situation.
+
+#### Transaction Submissions
+Certain transaction submission failures also need adjustments to succeed. If the XDR is malformed or the transaction is somehow otherwise invalid, you'll encounter a `400 Bad Request` (for example, when excluding a source account as seen in the [API reference example](../../api/errors/response/)). Both transactions and their requisite operations can easily be malformed: ook at the `extras.result_codes` field for details and cross-reference them with the appropriate [Result Codes documentation](../../api/errors/result-codes/) to determine specifics.
```

Comment:
```suggestion
Certain transaction submission failures also need adjustments to succeed. If the XDR is malformed or the transaction is somehow otherwise invalid, you'll encounter a `400 Bad Request` (for example, when excluding a source account as seen in the [API reference example](../../api/errors/response/)). Both transactions and their requisite operations can easily be malformed: look at the `extras.result_codes` field for details and cross-reference them with the appropriate [Result Codes documentation](../../api/errors/result-codes/) to determine specifics.
```